test/saveload.jl

```diff
@@ -21,6 +21,12 @@
         @test before[V(3)].fn == *
     end
 
+    @testset "MatMul" begin
+        ort_test(NNlib.batched_mul, rand(3, 4, 5), rand(4, 3))
```

Comment:
Is loading a 2D matmul -> * already tested somewhere else?

Reply:
Not really. I thought I would come up with a suitable solution while CI/CD runs these tests, but it turned out to be a bit harder than I expected :smile: 

The problem is that due to asymmetry we have no way to generate a graph with 2D MatMul. I have several ideas how to overcome it, but all of them have long-term downsides:

* Put a manually created graph with MatMul to the repo. Pros: relatively easy. Cons: doesn't test Julia -> ONNX -> Julia conversion.
* Introduce `ONNX.matmul()` operation. Pros: we can later extend it to 1D and ND cases. Cons: unlike `*` and `batched_mul()`, this `matmul` will be unique to ONNX.jl.
* Throw an error on 2Dx2D MatMul to indicate it's not tested. 
* Ignore the issue since it's actually tested manually :facepalm: 

So far adding `ONNX.matmul()` seems to be the easiest, but I need to sleep over this idea. Or hear some convincing feedback :)

Reply:
I like option 1, perhaps using the [Python API](https://github.com/onnx/onnx/blob/main/docs/PythonAPIOverview.md) for graph creation so that we don't have to save the actual graphs in the repo. This will be useful in future since the number of ops that can't be round-tripped will only increase.

Let's not hold up this PR because of that though. LMK when you want it pushed and I'll put in the approval.

Reply:
Python API is on the same level of abstraction as our primitives, so I just used them. The test I've got is quite verbose, maybe I'll see a way to simplify it later, but for now I'm glad it just works.